engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugOverlay.java

```diff
@@ -85,9 +88,17 @@ public Boolean get() {
             debugLine1.bindText(new ReadOnlyBinding<String>() {
                 @Override
                 public String get() {
+                    //Memory stats without using Runtime.getRuntime() for client side
+                    String memHeap="";
+                    for (MemoryPoolMXBean mpBean: ManagementFactory.getMemoryPoolMXBeans()) {
+                        if (mpBean.getType() == MemoryType.HEAP) {
+                            memHeap+="\n[Client] Memory Heap: "+mpBean.getName() +":"+ mpBean.getUsage();
+                        }
+                    }
                     double memoryUsage = ((double) Runtime.getRuntime().totalMemory() - (double) Runtime.getRuntime().freeMemory()) / 1048576.0;
-                    return String.format("FPS: %.2f, Memory Usage: %.2f MB, Total Memory: %.2f MB, Max Memory: %.2f MB",
-                            time.getFps(), memoryUsage, Runtime.getRuntime().totalMemory() / 1048576.0, Runtime.getRuntime().maxMemory() / 1048576.0);
+                    //showing server, client stats organized way.
+                    return String.format("[Server] Memory Usage: %.2f MB, Total Memory: %.2f MB, Max Memory: %.2f MB\n[Client] FPS: %.2f %s",
```

Comment:
I would extract this to a variable as is done for `memoryUsage` above and thus separate the "computation" from the String generation. Just a personal preference maybe ..